alpine/Dockerfile

```
@@ -97,7 +97,7 @@ RUN apk add --no-cache --virtual .build-deps \
     && cd /tmp \
     && if [ -n "${RESTY_EVAL_PRE_CONFIGURE}" ]; then eval $(echo ${RESTY_EVAL_PRE_CONFIGURE}); fi \
     && cd /tmp \
-    && curl -fSL https://www.openssl.org/source/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
+    && curl -fSL https://www.openssl.org/source/old/1.1.1/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
```

Hard-coding `1.1.1` here breaks things if `RESTY_OPENSSL_VERSION` is set to an older version. The path should be derived from the build arg. Similar issues exist with the patching of openssl done below, but that is outside the scope of this PR.

Another problem is that the latest openssl version (currently 1.1.1f) can't be referenced as that is still hosted directly under https://openssl.org/source. Maybe just expose a build arg for the openssl path with a default of https://www.openssl.org/source/old/?

```suggestion
    && curl -fSL https://www.openssl.org/source/old/$(echo $RESTY_OPENSSL_VERSION | sed 's/[a-z]$//')/openssl-${RESTY_OPENSSL_VERSION}.tar.gz -o openssl-${RESTY_OPENSSL_VERSION}.tar.gz \
```